### PR TITLE
[PM-11598] GitHub Release Workflow

### PR DESCRIPTION
**Pull request by vvolkgang**
## 🎟️ Tracking

https://bitwarden.atlassian.net/browse/PM-11598

## 📔 Objective

Adds a workflow to create github releases based off a build action Run, by creating a new GitHub Release with automated release notes, uploading the artifacts from the provided run ID and appending run URL to the release description.

:warning: Currently, version name and number have to be manually provided, this will be automated after build.yml is improved to upload a `version-info.json` with that info. 

Additionally, adds a release-notes.sh script that may be used for release branches with cherry picks directly pushed without a PR. 

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: .github/workflows/github-release.yml [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+          if [[ "$release_branch" != "main" && ! "$release_branch" =~ ^release/ ]]; then
+            echo "Error: Branch '$release_branch' is not 'main' or a release branch starting with 'release/'. Releases must be created from protected branches."
+            exit 1
+          fi
+
+
+          # if ! gh api "repos/${{ github.repository }}/branches/$release_branch/protection" | grep -q "required_status_checks"; then
+          #   echo "Error: Branch '$release_branch' is not protected. Releases must be created from protected branches."
+          #   exit 1
+          # fi
```
Comment (fedemkr) [inline .github/workflows/github-release.yml:60]: ❓ Should these comments remain here? I would guess it's the generic version to check if the release branch is protected but given the previous check wanted to raise if this should be removed.

**Review by fedemkr — COMMENTED** [review]
Looks good, just one small question.
Also, maybe I've missed it but should the `release-notes.sh` script take into account Co-authors?

**Comment by vvolkgang** [comment]
> Looks good, just one small question. Also, maybe I've missed it but should the `release-notes.sh` script take into account Co-authors?

💡 Co-authors would be a cool add! I'll leave it as is for now because I'm looking into other options but if we happen to stick with this script, we can add that support for that too.

File: .github/workflows/github-release.yml (hunk 1 above)
Comment (vvolkgang) [reply .github/workflows/github-release.yml]: Took the opportunity to improve this by adding an input to control which branch protection should be executed. This API call is the ideal solution but unfortunately requires a token with Organisation Administration read scopes, which most repos won't have by default.

**Review by SaintPatrck — APPROVED** [review]

**Pull request merged**